libs/libcommon/src/libcommon/orchestrator.py

```diff
@@ -689,9 +690,21 @@ def _create_plan(self) -> None:
             + list(self.cache_status.cache_is_job_runner_obsolete.values())
             + list(self.cache_status.cache_has_different_git_revision.values())
         )
+
+        @lru_cache
+        def is_big(config: str) -> bool:
+            num_bytes = get_num_bytes_from_config_infos(dataset=self.dataset, config=config)
+            if num_bytes is None:
+                return False
+            else:
+                return num_bytes > self.processing_graph.min_bytes_for_bonus_difficulty
+
         for artifact_state in artifact_states:
             valid_pending_jobs_df = artifact_state.job_state.valid_pending_jobs_df
             if valid_pending_jobs_df.empty:
+                difficulty = artifact_state.processing_step.difficulty
+                if isinstance(artifact_state.config, str) and is_big(config=artifact_state.config):
+                    difficulty += artifact_state.processing_step.bonus_difficulty_if_dataset_is_big
```

Comment:
maybe limit to 100

Reply:
```
            # increase difficulty according to number of failed runs
            difficulty = min(DEFAULT_DIFFICULTY_MAX, difficulty + self.failed_runs * DIFFICULTY_BONUS_BY_FAILED_RUNS)
```

^might be factorized too

Reply:
I'm changing the logic in AfterJobPlan in https://github.com/huggingface/datasets-server/pull/2523 so no need to factorize imo